### PR TITLE
[TT-9134]: added global headers to udg config

### PR DESCRIPTION
**Pull request by kofoworola**
## Description

This change introduces a new GlobalHeaders field to the graphql section of the APIDef, the headers set here are then set as then passed onto the various data sources

[TT-9134](https://tyktech.atlassian.net/browse/TT-9134)
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

[TT-9134]: https://tyktech.atlassian.net/browse/TT-9134?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5256/merge
Commit: ff8f85359caa3dcfe114c04938292dc26a3753bd 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5421029844)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5256/merge
Commit: 2c6573f80a50f3f737d971589f7ef3de491c6c53 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5422993813)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5256/merge
Commit: 488cf032c5d517f023c6bdd15423bbb5997a02a6 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5439286850)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5256/merge
Commit: e3edb2d6145ca4fe419a09948cbabf1215a10178 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5439282491)

**Review by rhianeKobar — APPROVED** [review]

File: apidef/api_definitions.go [hunk 1]
```diff
@@ -782,8 +782,9 @@ type GraphQLSubgraphEntity struct {
 }
 
 type GraphQLEngineConfig struct {
-	FieldConfigs []GraphQLFieldConfig      `bson:"field_configs" json:"field_configs"`
-	DataSources  []GraphQLEngineDataSource `bson:"data_sources" json:"data_sources"`
+	FieldConfigs  []GraphQLFieldConfig      `bson:"field_configs" json:"field_configs"`
+	DataSources   []GraphQLEngineDataSource `bson:"data_sources" json:"data_sources"`
+	GlobalHeaders map[string]string         `bson:"global_headers" json:"global_headers"`
```
Comment (pvormste) [inline apidef/api_definitions.go:787]: we also need to extend `schema.go`:
https://github.com/TykTechnologies/tyk/blob/master/apidef/schema.go

Comment (pvormste) [reply apidef/api_definitions.go:787]: Also `map[string]string` may not be the perfect type if we want to make this extensible for the future.
Better would be a struct:
```
GlobalHeaders  []UDGGlobalHeader

type UDGGlobalHeader struct {
  Key string
  Value string
  // Additional data in future, e.g.
  // EnabledForAll bool
}

// As JSON
// { "global_headers": [ { "key": "Authorization", "value": "ey123" } ]
```

or also possible (maybe better for frontend):
```
GlobalHeaders  map[string]UDGGlobalHeaderValue

type UDGGlobalHeaderValue struct {
  Value string
  // Additional data in future, e.g.
  // EnabledForAll bool
}

// As JSON
// { "global_headers": { "Authorization": { "value": "ey123" } }
```

**Review by pvormste — CHANGES_REQUESTED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5256/merge
Commit: cf6928c87ac1a64c6487ca021b535dc879aa6cb7 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5443782356)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5256/merge
Commit: 85a00b75d9239892851147752842a709be040172 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5443771997)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5256/merge
Commit: ce9e6b799ef5e7a76b2ade16349e81525b752957 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5444088118)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5256/merge
Commit: 5926e9dde434de3f0512c8c36eb7fbabb4d2623e 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5444121041)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5256/merge
Commit: 6ab0d7c1474d316e137cb8d01e3441b70f4f134a 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5444127743)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5256/merge
Commit: 3f13066c3e3dcc2be0a2055b62b0de9d1dfdd88c 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5455625852)

**Review by pvormste — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5256/merge
Commit: a63bcbcbefa7f96026bd1f1e20c563e5e031abad 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5463707328)

**Pull request merged**